json/test_simdjson.cpp

```diff
@@ -41,11 +40,9 @@ bool on_demand::run(const padded_string &json) {
   count = 0;
   auto doc = parser.iterate(json);
   for (ondemand::object point_object : doc["coordinates"]) {
-    for (auto field : point_object) {
-      if      (field.key() == "x") { my_point.x += double(field.value()); }
-      else if (field.key() == "y") { my_point.y += double(field.value()); }
-      else if (field.key() == "z") { my_point.z += double(field.value()); }
-    }
+    my_point.x += double(point_object["x"]);
```

Comment:
It's *possible* that you don't even need the double(). You generally don't need it for assignment, anyway!

Reply:
"It's possible that you don't even need the double(). You generally don't need it for assignment, anyway!"

I think you do here because double += '?' is defined for many '?' types.